### ISSUE TITLE
get_users on junos results in AttributeError: 'NoneType' object has no attribute 'strip'

### ISSUE THREAD

**Issue by ggiesen** [napalm-automation/napalm#1385]
### Description of Issue/Question
Calling get_users on junos results in an AttributeError: 'NoneType' object has no attribute 'strip'

This issue was closed in #1294 but according to the [Getters support matrix](https://napalm.readthedocs.io/en/latest/support/index.html#getters-support-matrix) , get_users should be supported on all platforms

### Did you follow the steps from https://github.com/napalm-automation/napalm#faq
(Place an ``x`` between the square brackets where applicable)

- [x] Yes
- [] No

### napalm version
(Paste verbatim output from `pip freeze | grep napalm` between quotes below)

```
napalm==3.2.0
```

### Network operating system version
(Paste verbatim output from `show version` - or equivalent - between quotes below)

```
Hostname: router
Model: mx204
Junos: 18.4R2.7
JUNOS OS Kernel 64-bit  [20190517.f0321c3_builder_stable_11]
JUNOS OS libs [20190517.f0321c3_builder_stable_11]
JUNOS OS runtime [20190517.f0321c3_builder_stable_11]
JUNOS OS time zone information [20190517.f0321c3_builder_stable_11]
JUNOS network stack and utilities [20190627.055352_builder_junos_184_r2]
JUNOS libs [20190627.055352_builder_junos_184_r2]
JUNOS OS libs compat32 [20190517.f0321c3_builder_stable_11]
JUNOS OS 32-bit compatibility [20190517.f0321c3_builder_stable_11]
JUNOS libs compat32 [20190627.055352_builder_junos_184_r2]
JUNOS runtime [20190627.055352_builder_junos_184_r2]
Junos vmguest package [20190627.055352_builder_junos_184_r2]
JUNOS sflow mx [20190627.055352_builder_junos_184_r2]
JUNOS py extensions [20190627.055352_builder_junos_184_r2]
JUNOS py base [20190627.055352_builder_junos_184_r2]
JUNOS OS vmguest [20190517.f0321c3_builder_stable_11]
JUNOS OS crypto [20190517.f0321c3_builder_stable_11]
JUNOS na telemetry [18.4R2.7]
JUNOS Security Intelligence [20190627.055352_builder_junos_184_r2]
JUNOS mx libs compat32 [20190627.055352_builder_junos_184_r2]
JUNOS mx runtime [20190627.055352_builder_junos_184_r2]
JUNOS common platform support [20190627.055352_builder_junos_184_r2]
JUNOS Openconfig [18.4R2.7]
JUNOS mtx network modules [20190627.055352_builder_junos_184_r2]
JUNOS modules [20190627.055352_builder_junos_184_r2]
JUNOS mx modules [20190627.055352_builder_junos_184_r2]
JUNOS mx libs [20190627.055352_builder_junos_184_r2]
JUNOS SQL Sync Daemon [20190627.055352_builder_junos_184_r2]
JUNOS mtx Data Plane Crypto Support [20190627.055352_builder_junos_184_r2]
JUNOS daemons [20190627.055352_builder_junos_184_r2]
JUNOS mx daemons [20190627.055352_builder_junos_184_r2]
JUNOS -MX appidd application-identification daemon [20190627.055352_builder_junos_184_r2]
JUNOS Services URL Filter package [20190627.055352_builder_junos_184_r2]
JUNOS Services TLB Service PIC package [20190627.055352_builder_junos_184_r2]
JUNOS Services Telemetry [20190627.055352_builder_junos_184_r2]
JUNOS Services TCP-LOG [20190627.055352_builder_junos_184_r2]
JUNOS Services SSL [20190627.055352_builder_junos_184_r2]
JUNOS Services SOFTWIRE [20190627.055352_builder_junos_184_r2]
JUNOS Services Stateful Firewall [20190627.055352_builder_junos_184_r2]
JUNOS Services RTCOM [20190627.055352_builder_junos_184_r2]
JUNOS Services RPM [20190627.055352_builder_junos_184_r2]
JUNOS Services PCEF package [20190627.055352_builder_junos_184_r2]
JUNOS Services NAT [20190627.055352_builder_junos_184_r2]
JUNOS Services Mobile Subscriber Service Container package [20190627.055352_builder_junos_184_r2]
JUNOS Services MobileNext Software package [20190627.055352_builder_junos_184_r2]
JUNOS Services Logging Report Framework package [20190627.055352_builder_junos_184_r2]
JUNOS Services LL-PDF Container package [20190627.055352_builder_junos_184_r2]
JUNOS Services Jflow Container package [20190627.055352_builder_junos_184_r2]
JUNOS Services Deep Packet Inspection package [20190627.055352_builder_junos_184_r2]
JUNOS Services IPSec [20190627.055352_builder_junos_184_r2]
JUNOS Services IDS [20190627.055352_builder_junos_184_r2]
JUNOS IDP Services [20190627.055352_builder_junos_184_r2]
JUNOS Services HTTP Content Management package [20190627.055352_builder_junos_184_r2]
JUNOS Services Flowd MS-MPC Software package [20190627.055352_builder_junos_184_r2]
JUNOS Services Crypto [20190627.055352_builder_junos_184_r2]
JUNOS Services Captive Portal and Content Delivery Container package [20190627.055352_builder_junos_184_r2]
JUNOS Services COS [20190627.055352_builder_junos_184_r2]
JUNOS AppId Services [20190627.055352_builder_junos_184_r2]
JUNOS Services Application Level Gateways [20190627.055352_builder_junos_184_r2]
JUNOS Services AACL Container package [20190627.055352_builder_junos_184_r2]
JUNOS SDN Software Suite [20190627.055352_builder_junos_184_r2]
JUNOS Extension Toolkit [20190627.055352_builder_junos_184_r2]
JUNOS Packet Forwarding Engine Support (wrlinux9) [20190627.055352_builder_junos_184_r2]
JUNOS Packet Forwarding Engine Support (MX/EX92XX Common) [20190627.055352_builder_junos_184_r2]
JUNOS Packet Forwarding Engine Support (M/T Common) [20190627.055352_builder_junos_184_r2]
JUNOS Packet Forwarding Engine Support (MX Common) [20190627.055352_builder_junos_184_r2]
JUNOS J-Insight [20190627.055352_builder_junos_184_r2]
JUNOS jfirmware [20190627.055352_builder_junos_184_r2]
JUNOS Online Documentation [20190627.055352_builder_junos_184_r2]
JUNOS jail runtime [20190517.f0321c3_builder_stable_11]
```

### Steps to Reproduce the Issue
$ python3 -m venv /opt/napalm/
$ source /opt/napalm/bin/activate
$ pip install pip -U
$ pip install napalm
$ napalm --user user -v junos router call get_users
Enter password: 
### Error Traceback
(Paste the complete traceback of the exception between quotes below)

```
2021-02-26 17:06:05,234 - napalm - ERROR - method - Failed: 'NoneType' object has no attribute 'strip'

================= Traceback =================

Traceback (most recent call last):
  File "/opt/napalm/bin/napalm", line 8, in <module>
    sys.exit(main())
  File "/opt/napalm/lib64/python3.6/site-packages/napalm/base/clitools/cl_napalm.py", line 308, in main
    run_tests(args)
  File "/opt/napalm/lib64/python3.6/site-packages/napalm/base/clitools/cl_napalm.py", line 291, in run_tests
    call_getter(device, args.method, **method_kwargs)
  File "/opt/napalm/lib64/python3.6/site-packages/napalm/base/clitools/cl_napalm.py", line 27, in wrapper
    r = func(*args, **kwargs)
  File "/opt/napalm/lib64/python3.6/site-packages/napalm/base/clitools/cl_napalm.py", line 255, in call_getter
    r = func(**kwargs)
  File "/opt/napalm/lib64/python3.6/site-packages/napalm/junos/junos.py", line 2140, in get_users
    root_user = self._get_root()
  File "/opt/napalm/lib64/python3.6/site-packages/napalm/junos/junos.py", line 2108, in _get_root
    root_items = root_table.items()
  File "/opt/napalm/lib64/python3.6/site-packages/jnpr/junos/factory/table.py", line 190, in items
    return list(zip(self.keys(), self.values()))
  File "/opt/napalm/lib64/python3.6/site-packages/jnpr/junos/factory/table.py", line 166, in keys
    self._key_list = self._keys()
  File "/opt/napalm/lib64/python3.6/site-packages/jnpr/junos/factory/table.py", line 146, in _keys
    return self._keys_simple(xpath + "/" + key_value)
  File "/opt/napalm/lib64/python3.6/site-packages/jnpr/junos/factory/table.py", line 114, in _keys_simple
    return [x.text.strip() for x in self.xml.xpath(xpath)]
  File "/opt/napalm/lib64/python3.6/site-packages/jnpr/junos/factory/table.py", line 114, in <listcomp>
    return [x.text.strip() for x in self.xml.xpath(xpath)]
AttributeError: 'NoneType' object has no attribute 'strip'
```

**Comment by ggiesen** [comment]
Note this is on CentOS 8, so this is not a Windows-specific issue

**Comment by ggiesen** [comment]
Can also confirm this happens on Junos boxes running 17.3R2.10 and 19.1R3-S3.2

**Comment by mirceaulinic** [comment]
Hi @ggiesen, would you be able to share the sanitized view of `show configuration system login | display xml` (not the values itself are important here, rather the structure) to see what's different?

**Comment by ggiesen** [comment]
At first I thought it might have been an issue with the use of custom login classes (and indeed, there is a separate issue where the priv level is reported as 0):

```
<rpc-reply xmlns:junos="http://xml.juniper.net/junos/18.2R1/junos">
    <configuration junos:commit-seconds="1612970592" junos:commit-localtime="2021-02-10 10:23:12 EST" junos:commit-user="rancid">
            <system>
                <login>
                    <idle-timeout>30</idle-timeout>
                    <class>
                        <name>network-engineering</name>
                        <permissions>all</permissions>
                    </class>
                    <class>
                        <name>network-provisioning</name>
                        <permissions>view</permissions>
                        <permissions>view-configuration</permissions>
                    </class>
                    <class>
                        <name>network-support</name>
                        <permissions>view</permissions>
                        <permissions>view-configuration</permissions>
                    </class>
                    <class>
                        <name>network-readonly</name>
                        <permissions>view</permissions>
                        <permissions>view-configuration</permissions>
                    </class>
                    <class>
                        <name>network-daemon</name>
                        <permissions>all</permissions>
                    </class>
                    <user>
                        <name>someuser</name>
                        <full-name>Some User</full-name>
                        <uid>2000</uid>
                        <class>super-user</class>
                        <authentication>
                            <encrypted-password>$6$jxvS9hZxZjpX5fVW$qBlCyilM/sB4PbodKwX4fOVTGklqZ26E6UaJKIXlUBIQz91WOmLT3wK.SRsrP1peQMxEBjIFw6eDc4zo/L4Rq1</encrypted-password>
                        </authentication>
                    </user>
                    <user>
                        <name>network-engineering</name>
                        <uid>2001</uid>
                        <class>network-engineering</class>
                    </user>
                    <user>
                        <name>network-provisioning</name>
                        <uid>2002</uid>
                        <class>network-provisioning</class>
                    </user>
                    <user>
                        <name>network-support</name>
                        <uid>2003</uid>
                        <class>network-support</class>
                    </user>
                    <user>
                        <name>network-readonly</name>
                        <uid>2004</uid>
                        <class>network-readonly</class>
                    </user>
                    <user>
                        <name>network-daemon</name>
                        <uid>2005</uid>
                        <class>network-daemon</class>
                    </user>
                </login>
            </system>
    </configuration>
    <cli>
        <banner></banner>
    </cli>
</rpc-reply>
```

but I simplied things using only standard login classes with the same result:

```
<rpc-reply xmlns:junos="http://xml.juniper.net/junos/18.4R2/junos">
    <configuration junos:commit-seconds="1614385660" junos:commit-localtime="2021-02-27 00:27:40 UTC" junos:commit-user="root">
            <system>
                <login>
                    <idle-timeout>30</idle-timeout>
                    <user>
                        <name>network-daemon</name>
                        <uid>2001</uid>
                        <class>super-user</class>
                    </user>
                    <user>
                        <name>network-engineering</name>
                        <uid>2002</uid>
                        <class>super-user</class>
                    </user>
                    <user>
                        <name>network-provisioning</name>
                        <uid>2003</uid>
                        <class>read-only</class>
                    </user>
                    <user>
                        <name>network-readonly</name>
                        <uid>2004</uid>
                        <class>read-only</class>
                    </user>
                    <user>
                        <name>network-support</name>
                        <uid>2005</uid>
                        <class>read-only</class>
                    </user>
                    <user>
                        <name>someuser</name>
                        <full-name>Some User</full-name>
                        <uid>2000</uid>
                        <class>super-user</class>
                        <authentication>
                            <encrypted-password>$6$jxvS9hZxZjpX5fVW$qBlCyilM/sB4PbodKwX4fOVTGklqZ26E6UaJKIXlUBIQz91WOmLT3wK.SRsrP1peQMxEBjIFw6eDc4zo/L4Rq1</encrypted-password>
                        </authentication>
                    </user>
                </login>
            </system>
    </configuration>
    <cli>
        <banner></banner>
    </cli>
</rpc-reply>
```

**Comment by ggiesen** [comment]
A napalm run with debugging enabled:

```
napalm --user user -v junos --debug router call get_users
Enter password: 
2021-03-08 01:08:07,896 - napalm - DEBUG - Starting napalm's debugging tool
2021-03-08 01:08:07,896 - napalm - DEBUG - Gathering napalm packages
2021-03-08 01:08:07,896 - napalm - DEBUG - napalm==3.2.0
2021-03-08 01:08:07,896 - napalm - DEBUG - get_network_driver - Calling with args: ('junos',), {}
2021-03-08 01:08:07,897 - napalm - DEBUG - get_network_driver - Successful
2021-03-08 01:08:07,897 - napalm - DEBUG - __init__ - Calling with args: (<class 'napalm.junos.junos.JunOSDriver'>, 'router', 
'user'), {'password': '*******', 'timeout': 60, 'optional_args': {}}
2021-03-08 01:08:07,897 - napalm - DEBUG - __init__ - Successful
2021-03-08 01:08:07,897 - napalm - DEBUG - pre_connection_tests - Calling with args: (<napalm.junos.junos.JunOSDriver object at 0x7f1d71f1a668>,), {}
2021-03-08 01:08:07,897 - napalm - DEBUG - open - Calling with args: (<napalm.junos.junos.JunOSDriver object at 0x7f1d71f1a668>,), {}
2021-03-08 01:08:08,459 - napalm - DEBUG - open - Successful
2021-03-08 01:08:08,459 - napalm - DEBUG - connection_tests - Calling with args: (<napalm.junos.junos.JunOSDriver object at 0x7f1d71f1a668>,), {}
2021-03-08 01:08:08,459 - napalm - DEBUG - get_facts - Calling with args: (<napalm.junos.junos.JunOSDriver object at 0x7f1d71f1a668>,), {}
2021-03-08 01:08:09,912 - napalm - DEBUG - Gathered facts:
{
    "vendor": "Juniper",
    "model": "MX204",
    "serial_number": "ZZ999",
    "os_version": "18.4R2.7",
    "hostname": "router",
    "fqdn": "router",
    "uptime": 45068206,
    "interface_list": [
        "gr-0/0/0",
        "ip-0/0/0",
        "lc-0/0/0",
        "lt-0/0/0",
        "mt-0/0/0",
        "pd-0/0/0",
        "pe-0/0/0",
        "pfe-0/0/0",
        "pfh-0/0/0",
        "ud-0/0/0",
        "ut-0/0/0",
        "vt-0/0/0",
        "xe-0/0/0:0",
        "xe-0/0/0:1",
        "xe-0/0/0:2",
        "xe-0/0/0:3",
        "xe-0/0/1:0",
        "xe-0/0/1:1",
        "xe-0/0/1:2",
        "xe-0/0/1:3",
        "xe-0/0/2:0",
        "xe-0/0/2:1",
        "xe-0/0/2:2",
        "xe-0/0/2:3",
        "xe-0/0/3:0",
        "xe-0/0/3:1",
        "xe-0/0/3:2",
        "xe-0/0/3:3",
        "xe-0/1/0",
        "xe-0/1/1",
        "xe-0/1/2",
        "xe-0/1/3",
        "xe-0/1/4",
        "xe-0/1/5",
        "xe-0/1/6",
        "xe-0/1/7",
        "cbp0",
        "demux0",
        "dsc",
        "em2",
        "em3",
        "em4",
        "esi",
        "fti0",
        "fti1",
        "fti2",
        "fti3",
        "fti4",
        "fti5",
        "fti6",
        "fti7",
        "fxp0",
        "gre",
        "ipip",
        "irb",
        "jsrv",
        "lo0",
        "lsi",
        "mtun",
        "pimd",
        "pime",
        "pip0",
        "pp0",
        "rbeb",
        "tap",
        "vtep"
    ]
}
{
    "vendor": "Juniper",
    "model": "MX204",
    "serial_number": "ZZ999",
    "os_version": "18.4R2.7",
    "hostname": "router",
    "fqdn": "router",
    "uptime": 45068206,
    "interface_list": [
        "gr-0/0/0",
        "ip-0/0/0",
        "lc-0/0/0",
        "lt-0/0/0",
        "mt-0/0/0",
        "pd-0/0/0",
        "pe-0/0/0",
        "pfe-0/0/0",
        "pfh-0/0/0",
        "ud-0/0/0",
        "ut-0/0/0",
        "vt-0/0/0",
        "xe-0/0/0:0",
        "xe-0/0/0:1",
        "xe-0/0/0:2",
        "xe-0/0/0:3",
        "xe-0/0/1:0",
        "xe-0/0/1:1",
        "xe-0/0/1:2",
        "xe-0/0/1:3",
        "xe-0/0/2:0",
        "xe-0/0/2:1",
        "xe-0/0/2:2",
        "xe-0/0/2:3",
        "xe-0/0/3:0",
        "xe-0/0/3:1",
        "xe-0/0/3:2",
        "xe-0/0/3:3",
        "xe-0/1/0",
        "xe-0/1/1",
        "xe-0/1/2",
        "xe-0/1/3",
        "xe-0/1/4",
        "xe-0/1/5",
        "xe-0/1/6",
        "xe-0/1/7",
        "cbp0",
        "demux0",
        "dsc",
        "em2",
        "em3",
        "em4",
        "esi",
        "fti0",
        "fti1",
        "fti2",
        "fti3",
        "fti4",
        "fti5",
        "fti6",
        "fti7",
        "fxp0",
        "gre",
        "ipip",
        "irb",
        "jsrv",
        "lo0",
        "lsi",
        "mtun",
        "pimd",
        "pime",
        "pip0",
        "pp0",
        "rbeb",
        "tap",
        "vtep"
    ]
}
2021-03-08 01:08:09,912 - napalm - DEBUG - get_facts - Successful
2021-03-08 01:08:09,913 - napalm - DEBUG - method - Calling with args: (<napalm.junos.junos.JunOSDriver object at 0x7f1d71f1a668>, 'get_users'), {}
2021-03-08 01:08:09,913 - napalm - DEBUG - get_users - Attempting to resolve method
2021-03-08 01:08:09,913 - napalm - DEBUG - get_users - Attempting to call method with kwargs: {}
2021-03-08 01:08:10,132 - napalm - DEBUG - get_users - Response
{
    "network-daemon": {
        "level": 15,
        "password": "",
        "sshkeys": []
    },
    "network-engineering": {
        "level": 15,
        "password": "",
        "sshkeys": []
    },
    "network-provisioning": {
        "level": 1,
        "password": "",
        "sshkeys": []
    },
    "network-readonly": {
        "level": 1,
        "password": "",
        "sshkeys": []
    },
    "network-support": {
        "level": 1,
        "password": "",
        "sshkeys": []
    },
    "some-user": {
        "level": 15,
        "password": "$6$jxvS9hZxZjpX5fVW$qBlCyilM/sB4PbodKwX4fOVTGklqZ26E6UaJKIXlUBIQz91WOmLT3wK.SRsrP1peQMxEBjIFw6eDc4zo/L4Rq1",
        "sshkeys": []
    },
    "root": {
        "level": 20,
        "password": "$6$jxvS9hZxZjpX5fVW$qBlCyilM/sB4PbodKwX4fOVTGklqZ26E6UaJKIXlUBIQz91WOmLT3wK.SRsrP1peQMxEBjIFw6eDc4zo/L4Rq1",
        "sshkeys": []
    }
}
2021-03-08 01:08:10,132 - napalm - DEBUG - method - Successful
2021-03-08 01:08:10,132 - napalm - DEBUG - close - Calling with args: (<napalm.junos.junos.JunOSDriver object at 0x7f1d71f1a668>,), {}      
2021-03-08 01:08:10,235 - napalm - DEBUG - close - Successful
2021-03-08 01:08:10,235 - napalm - DEBUG - post_connection_tests - Calling with args: (<napalm.junos.junos.JunOSDriver object at 0x7f1d71f1a668>,), {}
```

**Comment by minefuto** [comment]
Hi @mirceaulinic,

I reproduced this issue on MX480/16.1R1-S3.
In my opinion, the cause is "key: root-authentication" in junos_root_table.
When junos-eznc gets the key's value, it may return `NoneType` instead of a `""`(empty string),
but I don't know why this occurs(It seems there is a difference in the XML from the router).

If we want to fix this problem on the napalm side, I think we need to change a key where junos-eznc does not return `NoneType`.
I think it is effective to change the key to `root-authentication/encrypted-password` because JUNOS must set a root password.
Is there any concerns?

normal xml pattern:
```
<configuration xmlns="http://xml.juniper.net/xnm/1.1/xnm" xmlns:junos="http://xml.juniper.net/junos/16.1R3/junos" xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" changed-seconds="1615795490" changed-localtime="2021-03-15 17:04:50 JST"><system><root-authentication><encrypted-password>$5$PTqcSxfm$uUDOcNfKDi9KG6KfqMQShr0xY2ltUkKzTbhnS.hosW1</encrypted-password>
        </root-authentication>
        <login><user><name>test</name>
                <uid>2003</uid>
                <class>super-user</class>
                <authentication><encrypted-password>$5$oKzr/2sw$rgw0vgX0qZ/HGEKzBM6Zz914HMiqJJtEz9StWa.wR3.</encrypted-password>
                </authentication>
            </user>
        </login>
        <services><ftp></ftp>
            <ssh></ssh>
            <telnet></telnet>
        </services>
    </system>
</configuration>
```

error xml pattern:
```
<configuration changed-seconds="1615795490" changed-localtime="2021-03-15 17:04:50 JST"><system><root-authentication><encrypted-password>$5$PTqcSxfm$uUDOcNfKDi9KG6KfqMQShr0xY2ltUkKzTbhnS.hosW1</encrypted-password></root-authentication><login><user><name>test</name><uid>2003</uid><class>super-user</class><authentication><encrypted-password>$5$oKzr/2sw$rgw0vgX0qZ/HGEKzBM6Zz914HMiqJJtEz9StWa.wR3.</encrypted-password></authentication></user></login><services><ftp>
            </ftp><ssh>
            </ssh><telnet>
            </telnet></services></system></configuration>
```

**Comment by mirceaulinic** [comment]
@ggiesen would you be able to check @minefuto's fix from #1409 and confirm it solves your issue?

**Comment by ggiesen** [comment]
I can confirm @minefuto's fix in #1409 solves the issue.

```
napalm --user user -v junos --debug router call get_users
Enter password: 
{
    "network-daemon": {
        "level": 15,
        "password": "",
        "sshkeys": []
    },
    "network-engineering": {
        "level": 15,
        "password": "",
        "sshkeys": []
    },
    "network-provisioning": {
        "level": 1,
        "password": "",
        "sshkeys": []
    },
    "network-readonly": {
        "level": 1,
        "password": "",
        "sshkeys": []
    },
    "network-support": {
        "level": 1,
        "password": "",
        "sshkeys": []
    },
    "some-user": {
        "level": 15,
        "password": "$6$jxvS9hZxZjpX5fVW$qBlCyilM/sB4PbodKwX4fOVTGklqZ26E6UaJKIXlUBIQz91WOmLT3wK.SRsrP1peQMxEBjIFw6eDc4zo/L4Rq1",
        "sshkeys": []
    },
    "root": {
        "level": 20,
        "password": "$6$jxvS9hZxZjpX5fVW$qBlCyilM/sB4PbodKwX4fOVTGklqZ26E6UaJKIXlUBIQz91WOmLT3wK.SRsrP1peQMxEBjIFw6eDc4zo/L4Rq1",
        "sshkeys": []
    }
}
```
